src/resources/filters/quarto-pre/line-numbers.lua

```diff
@@ -1,29 +1,35 @@
 -- line-numbers.lua
 -- Copyright (C) 2020-2022 Posit Software, PBC
 
+local constants = require("modules/constants")
 
 function line_numbers()
   return {
     CodeBlock = function(el)
       if #el.attr.classes > 0 then
         local lineNumbers = lineNumbersAttribute(el)
+        el.attr.attributes[constants.kCodeLineNumbers] = nil
         if lineNumbers ~= false then
           -- use the pandoc line numbering class
           el.attr.classes:insert("number-lines")
-          -- remove for all formats except reveal
-          if not _quarto.format.isRevealJsOutput() then
-            el.attr.attributes["code-line-numbers"] = nil
+          -- remove for all formats except reveal and docusaurus
+          if type(lineNumbers) == "string" and (_quarto.format.isRevealJsOutput() or _quarto.format.isDocusaurusOutput()) then
+            if _quarto.format.isDocusaurusOutput() and lineNumbers:match("|") then
+              -- docusaurus does not support step separator like revealjs
+              lineNumbers = lineNumbers:gsub("|", ",")
```

Comment:
Since the Docusaurus syntax comes from here and is quite different, it may not be necessary to do a conversion from the revealjs syntax.
https://www.npmjs.com/package/parse-numeric-range

Reply:
I'm in favor or removing the step separator and supporting Reveal syntax when possible. In the long run it would be nice to parse the range expression and be able to generate a compatible syntax for the various formats that we're targeting (e.g. have a `quarto syntax` for ranges which will emit reveal, Docusaurus, material, etc...)

Reply:
> In the long run it would be nice to parse the range expression and be able to generate a compatible syntax for the various formats that we're targeting (e.g. have a `quarto syntax` for ranges which will emit reveal, Docusaurus, material, etc...)

It's definitely the best! Thanks.